### ISSUE TITLE
Need help in the understanding of the algorithm and maybe there is some improvements

### ISSUE THREAD

**Issue by lhprojects** [NicolasHug/Surprise#372]
#### Description

I'm learning the FunkSVD recently. I think https://github.com/NicolasHug/Surprise/blob/46b9914995e6c8c7d227b46f2eaeef2d4600580f/surprise/prediction_algorithms/matrix_factorization.pyx#L18
implemented the algorithm described in 
https://sifter.org/~simon/journal/20061211.html (+ bias) and https://surprise.readthedocs.io/en/stable/matrix_factorization.html#surprise.prediction_algorithms.matrix_factorization.SVD

The update rule of `pu` and `qi` in the code is famous and should work.
However, the description in https://surprise.readthedocs.io/en/stable/matrix_factorization.html#surprise.prediction_algorithms.matrix_factorization.SVD
and the comment in the source code are really confusing for me.

You said the update rule comes from the SGD for the loss with a L2 regularization.
But strictly speaking, for each example (rating), *you should shrink the `pu`  for every user, and the `qi` for every item*.
But the code only shrinks the `pu` and `qi` for the user and the item of the example.
Thus the effective $\lambda$ of regularization will be larger for a user/item that occurs more often in the training set, and it will be zero for a user/item not in the training set.

Maybe I should run some examples, but the current code should work, just the doc let me fell confused.

#### Versions

import platform; print(platform.platform())
import sys; print("Python", sys.version)
import surprise; print("surprise", surprise.__version__)
